### PR TITLE
Add airquality example cc3220 port

### PR DESCRIPTION
**Pull request by tcave-bb**
> Port of the air-quality-sensor-app to cc3220
>

### CODE REVIEW

File: examples/lock-app/cc32xx/main/AppTask.cpp
Comment (tobiasgraf) [inline examples/lock-app/cc32xx/main/AppTask.cpp:1]: Could you please provide some insight into why this change is being made? I’m curious to understand its connection to the air quality sensor app. Appreciate your clarification!

Comment (tcave-bb) [reply examples/lock-app/cc32xx/main/AppTask.cpp:1]: You re correct. This change should not have been committed. It has been removed.

File: examples/air-quality-sensor-app/air-quality-sensor-common/include/air-quality-sensor-manager.h [hunk 1]
```diff
@@ -1,11 +1,9 @@
-#include <app-common/zap-generated/ids/Attributes.h>
-#include <app-common/zap-generated/ids/Clusters.h>
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/include/air-quality-sensor-manager.h:2]: Why are the id includes being removed?  This file uses those ids directly, so it should include the headers they are defined in; basic include-what-you-use.

File: examples/air-quality-sensor-app/air-quality-sensor-common/include/relative-humidity-sensor-manager.h [hunk 2]
```diff
@@ -25,7 +25,6 @@ class RelativeHumiditySensorManager
         EmberAfStatus status = RelativeHumidityMeasurement::Attributes::MeasuredValue::Set(mEndpointId, newValue);
         VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == status,
                        ChipLogError(NotSpecified, "Failed to set RelativeHumidityMeasurement MeasuredValue attribute"));
-        ChipLogDetail(NotSpecified, "The new RelativeHumidityMeasurement value: %d", newValue);
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/include/relative-humidity-sensor-manager.h:28]: Why are these detail logs being removed, here and in other files?  This seems totally unrelated to the new example.

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/AirQualitySensorAttrUpdateHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+AirQualitySensorAttrUpdateHandler * AirQualitySensorAttrUpdateHandler::FromJSON(const char * json)
+{
+    Json::Reader reader;
+    Json::Value value;
+
+    if (!reader.parse(json, value))
+    {
+        ChipLogError(NotSpecified,
+                     "AllClusters App: Error parsing JSON with error %s:", reader.getFormattedErrorMessages().c_str());
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/AirQualitySensorAttrUpdateHandler.cpp:37]: This is not "AllClusters App".  This applies to a bunch of these logs.

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp [hunk 4]
```diff
@@ -136,78 +137,77 @@ void AirQualitySensorManager::Init()
...
-    ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", newValue);
+    Status status = mAirQualityInstance.UpdateAirQuality(static_cast<AirQualityEnum>(newValue));
+    VerifyOrReturn(Status::Success == status, ChipLogError(NotSpecified, "Failed to set AirQuality attribute"));
 }
 
 void AirQualitySensorManager::OnCarbonDioxideMeasurementChangeHandler(float newValue)
 {
-    mCarbonDioxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
-    ChipLogDetail(NotSpecified, "Updated Carbon Dioxide: %f", newValue);
+    ChipError chipError = mCarbonDioxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:146]:
```suggestion
    CHIP_ERROR chipError = mCarbonDioxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
```

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp [hunk 5]
```diff
@@ -136,78 +137,77 @@ void AirQualitySensorManager::Init()
 
 void AirQualitySensorManager::OnAirQualityChangeHandler(AirQualityEnum newValue)
 {
-    mAirQualityInstance.UpdateAirQuality(static_cast<AirQualityEnum>(newValue));
-    ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", newValue);
+    Status status = mAirQualityInstance.UpdateAirQuality(static_cast<AirQualityEnum>(newValue));
+    VerifyOrReturn(Status::Success == status, ChipLogError(NotSpecified, "Failed to set AirQuality attribute"));
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:141]: Worth logging the actual status/errors here and below?

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp [hunk 6]
```diff
@@ -136,78 +137,77 @@ void AirQualitySensorManager::Init()
...
-    ChipLogDetail(NotSpecified, "Updated Carbon Dioxide: %f", newValue);
+    ChipError chipError = mCarbonDioxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
+    VerifyOrReturn(CHIP_NO_ERROR == chipError, ChipLogError(NotSpecified, "Failed to set CarbonDioxide MeasuredValue attribute"));
 }
 
 void AirQualitySensorManager::OnCarbonMonoxideMeasurementChangeHandler(float newValue)
 {
-    mCarbonMonoxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
-    ChipLogDetail(NotSpecified, "Updated Carbon Monoxide value: %f", newValue);
+    ChipError chipError = mCarbonMonoxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:152]:
```suggestion
    CHIP_ERROR chipError = mCarbonMonoxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
```
and various other places in this file.

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp [hunk 7]
```diff
@@ -136,78 +137,77 @@ void AirQualitySensorManager::Init()
...
+    VerifyOrReturn(CHIP_NO_ERROR == chipError, ChipLogError(NotSpecified, "Failed to set CarbonMonoxide MeasuredValue attribute"));
 }
 
 void AirQualitySensorManager::OnNitrogenDioxideMeasurementChangeHandler(float newValue)
 {
-    mNitrogenDioxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
-    ChipLogDetail(NotSpecified, "Updated Nitrogen Dioxide value: %f", newValue);
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:158]: Why are all these detail logs logging the new value being removed?

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:1]: In general, why are all these cross-platform example changes part of an "add a new platform example" PR?  They should be in a separate, PR with a clear title and a clear description of the reasons for the changes.

File: examples/air-quality-sensor-app/cc32xx/README.md [hunk 8]
```diff
@@ -0,0 +1,232 @@
+# Matter `CC32XXSF` Lock Example Application
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/cc32xx/README.md:1]: This is very much not a lock.

I'm going to stop here.  Please check all the copy/paste to make sure it's actually correct.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/air-quality-sensor-app/air-quality-sensor-common/include/relative-humidity-sensor-manager.h (hunk 2 above)
Comment (tcave-bb) [reply examples/air-quality-sensor-app/air-quality-sensor-common/include/relative-humidity-sensor-manager.h:28]: You are correct, it is not related to the new example. I got carried away with the changes. I was adding error logs and at the same time noticed 3 levels of printing the same info (JSON parser, here and in the cluster layer).

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp (hunk 7 above)
Comment (tcave-bb) [reply examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:158]: See answer above https://github.com/project-chip/connectedhomeip/pull/29329/files#r1335751303

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/AirQualitySensorAttrUpdateHandler.cpp [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        return nullptr;
+    }
+
+    return Platform::New<AirQualitySensorAttrUpdateHandler>(std::move(value));
+}
+
+void AirQualitySensorAttrUpdateHandler::HandleCommand(intptr_t context)
+{
+    auto * self             = reinterpret_cast<AirQualitySensorAttrUpdateHandler *>(context);
+    std::string clusterName = self->mJsonValue["Name"].asString();
```
Comment (andy31415) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/AirQualitySensorAttrUpdateHandler.cpp:59]: Do we want to place common code that uses std::string and json? 

Concern is that we tried to avoid std::string/list/map and other things using heap from being used in embedded and examples end up being copied & pasted as real products, hence my concern of using JSON and stl here. It seemed fine in the original linux side because linux/darwin/android are large systems and they are ok with such things, however embedded seems more sensitive.

File: examples/air-quality-sensor-app/air-quality-sensor-common/include/air-quality-sensor-manager.h [hunk 10]
```diff
@@ -170,6 +170,7 @@ class AirQualitySensorManager
         mRadonConcentrationMeasurementInstance(mEndpointId, RadonConcentrationMeasurement::Id,
                                                ConcentrationMeasurement::MeasurementMediumEnum::kAir,
                                                ConcentrationMeasurement::MeasurementUnitEnum::kPpm),
+
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/include/air-quality-sensor-manager.h:173]: Why this random change to this file?

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp [hunk 11]
```diff
@@ -136,78 +137,99 @@ void AirQualitySensorManager::Init()
 
 void AirQualitySensorManager::OnAirQualityChangeHandler(AirQualityEnum newValue)
 {
-    mAirQualityInstance.UpdateAirQuality(static_cast<AirQualityEnum>(newValue));
-    ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", newValue);
+    Status status = mAirQualityInstance.UpdateAirQuality(static_cast<AirQualityEnum>(newValue));
+    VerifyOrReturn(
+        Status::Success == status,
+        ChipLogError(NotSpecified, "AirQualitySensor App: Failed to set AirQuality attribute %x", (unsigned int) status));
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:143]:
```suggestion
        ChipLogError(NotSpecified, "AirQualitySensor App: Failed to set AirQuality attribute %x", to_underlying(status)));
```

File: examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp [hunk 12]
```diff
@@ -136,78 +137,99 @@ void AirQualitySensorManager::Init()
...
 }
 
 void AirQualitySensorManager::OnCarbonDioxideMeasurementChangeHandler(float newValue)
 {
-    mCarbonDioxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
-    ChipLogDetail(NotSpecified, "Updated Carbon Dioxide: %f", newValue);
+    CHIP_ERROR chipError = mCarbonDioxideConcentrationMeasurementInstance.SetMeasuredValue(MakeNullable(newValue));
+    VerifyOrReturn(CHIP_NO_ERROR == chipError,
+                   ChipLogError(NotSpecified, "AirQualitySensor App: Failed to set CarbonDioxide MeasuredValue attribute %s",
+                                chip::ErrorStr(chipError)));
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:151]:
```suggestion
                   ChipLogError(NotSpecified, "AirQualitySensor App: Failed to set CarbonDioxide MeasuredValue attribute %" CHIP_ERROR_FORMAT,
                                chipError.Format());
```
And similar in other places.

File: examples/air-quality-sensor-app/cc32xx/main/AppTask.cpp [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+    CHIPDeviceManager & deviceMgr = CHIPDeviceManager::GetInstance();
+    ret                           = deviceMgr.Init(&EchoCallbacks);
+    if (ret != CHIP_NO_ERROR)
+    {
+        PLAT_LOG("CHIPDeviceManager::Init() failed: %s", ErrorStr(ret));
+        while (1)
+            ;
+    }
+
+    AirQualitySensorManager::InitInstance(EndpointId(AIR_QUALITY_SENSOR_ENDPOINT));
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/cc32xx/main/AppTask.cpp:166]: But the Matter thread is running by now, no?  The init here is going to race with that thread, likely causing memory corruption and whatnot....  This needs to either happen before the Matter thread spins up, or from the Matter thread, or synchronized via the Matter lock, I would think.

File: examples/air-quality-sensor-app/cc32xx/main/Logging.cpp [hunk 14]
```diff
@@ -0,0 +1,142 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/cc32xx/main/Logging.cpp:3]: Is this really copyright 2020?  In general, please check the copyright notices in all these files.

File: examples/air-quality-sensor-app/cc32xx/main/UartHelper.c [hunk 15]
```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2016, Texas Instruments Incorporated
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/cc32xx/main/UartHelper.c:2]: Should this file not be using the Matter license/copyright notice?

File: examples/air-quality-sensor-app/cc32xx/main/UartHelper.h
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/cc32xx/main/UartHelper.h:1]: This file needs a copyright notice, no?

File: examples/air-quality-sensor-app/linux/AirQualitySensorAppAttrUpdateDelegate.cpp
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/linux/AirQualitySensorAppAttrUpdateDelegate.cpp:1]: I tried to make this clear last time, but just to try again: There should probably be one PR to factor out this code into the "common" bits.  This would be much smaller than this PR, and much more straightforward to review.  Ideally a `git diff --color-moved` would show that it's all just code moving.

Then a separate PR to add the new platform-specific code.  This would _also_ be easier to review, because it would clearly be platform-specific.
